src/components/Messages.vue

```diff
@@ -0,0 +1,174 @@
+<template>
+  <div>
+    <div 
+      class="chat"
+      ref="chat"
+      v-if="messages"
+    >
+      <span
+        v-for="(message, index) in messages"
+        :key="index"
+        class="chat__message"
+        :data-sender="message.sender"
+      >
+        {{message.content}}
+        <i>{{convertDate(message.created.toDate())}}</i>
+      </span>
+      <strong>Last Updated: {{ convertDate(lastUpdate) }}</strong>
+    </div>
+    <form action="javascript:void(0)" class="chat__send">
+      <textarea
+        v-model="message"
+        required
+        placeholder="Message here..."
+        @keydown.enter.exact.prevent
+        @keyup.enter.exact="sendMessage"
+        @keydown.enter.shift.exact="newline"
+      ></textarea>
+      <button
+        class="button"
+        @click="sendMessage"
+      >Send Message</button>
+    </form>
+  </div>
+</template>
+
+<script>
+import API from '../api';
+import utilities from '../utilities';
+
+export default {
+  props: {
+    caseId: {
+      type: String,
+      default: null
+    },
+  },
+  data() {
+    return {
+      message: "",
+      messages: [],
+      lastUpdate: "",
+    }
+  },
+  created() {
+    this.getMessages()
+  },
+  watch: {
+    messages(newValue, oldValue) {
+      setTimeout(() => {
+        this.$refs.chat.scrollTop = 9999999999;
+      }, 300)
+    }
+  },
+  methods: {
+    convertDate(date) {
+      return utilities.dateMapper(date)
+    },
+    getMessages() {
+      this.messages = [];
+      this.lastUpdate = new Date();
+      API.messages.get(this.$props.caseId)
```

Comment:
Maybe this could be a good place to [listen to real-time updates](https://firebase.google.com/docs/firestore/query-data/listen).

If anyone, anywhere sends a message this would automatically update the message list without refreshing the view. 

What do you think?

Reply:
https://github.com/anonymouslyreportingtool/webapp/pull/7/commits/d4973d50d074beabfdc285a51ef87790845856f4 ✅